### PR TITLE
Added Support for Retrieving Nodes that support Mesh Capable Service

### PR DESCRIPTION
**Pull request by sammychinedu2ky**
This pull request adds support for this 
https://developer.hashicorp.com/consul/api-docs/catalog#list-nodes-for-mesh-capable-service

### CODE REVIEW

File: Consul/Catalog.cs [hunk 1]
```diff
@@ -237,6 +238,44 @@ public Task<QueryResult<CatalogService[]>> Service(string service, string tag, Q
             return req.Execute(ct);
         }
 
+        /// <summary>
+        /// Returns the nodes providing a mesh-capable service in a given datacenter. 
+        /// </summary>
+        /// <param name="service">The service ID</param>
+        /// <param name="ct">Cancellation token for long poll request. If set, OperationCanceledException will be thrown if the request is cancelled before completing</param>
+        /// <returns></returns>
```
Comment (winstxnhdw) [inline Consul/Catalog.cs:246]:
```suggestion
        /// <returns>A list of service instances</returns>
```

File: Consul/Catalog.cs [hunk 2]
```diff
@@ -237,6 +238,44 @@ public Task<QueryResult<CatalogService[]>> Service(string service, string tag, Q
...
+            return NodesForMeshCapableService(service, QueryOptions.Default, null, ct);
+        }
+
+        /// <summary>
+        /// Returns the nodes providing a mesh-capable service in a given datacenter.
+        /// </summary>
+        /// <param name="service">The service ID</param>
+        /// <param name="filter">Specifies the expression used to filter the queries results prior to returning the data</param>
+        /// <param name="ct">Cancellation token for long poll request. If set, OperationCanceledException will be thrown if the request is cancelled before completing</param>
+        /// <returns></returns>
```
Comment (winstxnhdw) [inline Consul/Catalog.cs:258]:
```suggestion
        /// <returns>A list of service instances</returns>
```

File: Consul/Catalog.cs [hunk 3]
```diff
@@ -237,6 +238,44 @@ public Task<QueryResult<CatalogService[]>> Service(string service, string tag, Q
...
+        }
+
+        /// <summary>
+        /// Returns the nodes providing a mesh-capable service in a given datacenter.
+        /// </summary>
+        /// <param name="service">The service ID</param>
+        /// <param name="q">Customized Query options</param>
+        /// <param name="filter">Specifies the expression used to filter the queries results prior to returning the data</param>
+        /// <param name="ct">Cancellation token for long poll request. If set, OperationCanceledException will be thrown if the request is cancelled before completing</param>
+        /// <returns></returns>
```
Comment (winstxnhdw) [inline Consul/Catalog.cs:271]:
```suggestion
        /// <returns>A list of service instances</returns>
```

**Review by winstxnhdw — COMMENTED** [review]
I didn't actually review it but I think these should be filled up :D

File: Consul/Catalog.cs (hunk 2 above)
Comment (sammychinedu2ky) [reply Consul/Catalog.cs:258]: thanks man

**Review by marcin-krystianc — COMMENTED** [review]
Hi @sammychinedu2ky,

the implementation seems very reasonable, but I dislike that we cannot create mesh-capable services with our library. Without it, the test scenario is kind of artificial, because we are not testing whether the endpoint actually returns anything.
Would you want to investigate more on mesh-capable services and what changes we would need to be able to create them with our library?

**Comment by sammychinedu2ky** [comment]
> Hi @sammychinedu2ky,
> 
> the implementation seems very reasonable, but I dislike that we cannot create mesh-capable services with our library. Without it, the test scenario is kind of artificial, because we are not testing whether the endpoint actually returns anything. Would you want to investigate more on mesh-capable services and what changes we would need to be able to create them with our library?

Alright Chief

**Comment by sammychinedu2ky** [comment]
The mesh service focuses more on Kubernetes support, which the test environment doesn't create

**Comment by sammychinedu2ky** [comment]
@marcin-krystianc .. please can you review this one first

**Comment by sammychinedu2ky** [comment]
Resolved by https://github.com/G-Research/consuldotnet/pull/273

**Pull request closed without merging**